dask/tests/test_delayed.py

```diff
@@ -855,3 +856,28 @@ def add(x, y):
     assert add.__name__ == "add"
     assert add.__doc__ == "This is a docstring"
     assert add.__wrapped__(1, 2) == 3
+
+
+def test_delayed_fusion():
+    @delayed
+    def test(i):
+        return i + 1
+
+    @delayed
+    def test2(i):
+        return i + 2
+
+    @delayed
+    def test3(i):
+        return i + 3
+
+    obj = test3(test2(test(10)))
+    dsk = dict(collections_to_dsk([obj]))
+    assert len(dsk) == 3
+
+    obj2 = test3(test2(test(10)))
```

Comment:
Non-blocking: I don't think I understand why `obj2` is needed for this test.

Reply:
I'd rather be overly careful even though it shouldn't matter. Calling anything on a delayed object could have side-effects (even though it shouldn't here)